tests/Unit/TestHelpers.hpp

```diff
@@ -141,46 +141,71 @@ void test_move_semantics(T&& a, const T& comparison) {
 // Test for iterators
 template <typename Container>
 void test_iterators(Container& c) {
-  CHECK(c.end() - c.begin() == c.size());
+  // CHECK(c.end() - c.begin() == c.size());
+  auto it = c.begin();
+  auto nx = std::next(it, std::distance(c.begin(), c.end()));
+  CHECK(nx == c.end());
```

Comment:
Why was the check against `c.size()` removed?  I don't think the new version tests much beyond that the `std::distance` call doesn't segfault or something.